react/src/lib/components/DeckGLMap/components/NorthArrow3D.tsx

```diff
@@ -0,0 +1,87 @@
+import DeckGL from "@deck.gl/react";
+import React from "react";
+import { ScenegraphLayer } from "@deck.gl/mesh-layers";
+import { COORDINATE_SYSTEM } from "deck.gl";
+import { OrbitView } from "@deck.gl/core";
+import { load } from "@loaders.gl/core";
+import { GLTFLoader } from "@loaders.gl/gltf";
+
+import { GLTF_JSON_NORTH_ARROW } from "../utils/northArrow";
+
+interface NorthArrow3DProps {
+    rotationOrbit: number;
+    rotationX: number;
+    position?: number[] | null;
+}
+
+const view = new OrbitView({
+    id: "main",
+    controller: false,
+    x: "0%",
+    y: "0%",
+    width: "100%",
+    height: "100%",
+});
+
+const INITIAL_VIEW_STATE = {
+    target: [0.0, 0.0],
+    zoom: 0,
+};
+
+const text = JSON.stringify(GLTF_JSON_NORTH_ARROW);
+const data = new Blob([text], { type: "text/plain" });
+const url = window.URL.createObjectURL(data);
+
+const NorthArrow3D: React.FC<NorthArrow3DProps> = ({
+    rotationOrbit,
+    rotationX,
+    position,
+}: NorthArrow3DProps) => {
+    const view_state = {
+        rotationOrbit: rotationOrbit + 90.0,
+        rotationX,
+        target: [0.0, 0.0],
+        zoom: 0,
+    };
+
+    const layers = [
+        new ScenegraphLayer({
+            id: "north-arrow-scenegraph-layer",
+            data: [{}],
+            scenegraph: load(url, GLTFLoader, {}),
+            getOrientation: [0.0, 0.0, 0.0],
+            getTranslation: [0.0, 0.0, 0.0],
+            getScale: [1, 1, 1],
+            sizeScale: 200, // 200
+            _lighting: "pbr",
+            pickable: false,
+            coordinateSystem: COORDINATE_SYSTEM.CARTESIAN,
+        }),
+    ];
+
+    return (
+        <DeckGL
```

Comment:
By adding another DeckGl component, you are consuming another GL context. In the web browser this is limited to 16. In the BLA dashboard we already consume several GL contexts (one per module). We should try to share the GL context within the map component. See https://deck.gl/docs/api-reference/core/deck#gl